url47ebb27197_asirbu_CorrectiveEvolution_tgz-pJ8-eu_fbk_soa_evolution_engine_impl_test_ProblemToSMVFuncTestJ8/src/eu/fbk/soa/process/node/ProcessNode.java

```diff
@@ -0,0 +1,38 @@
+package eu.fbk.soa.process.node;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlAttribute;
+
+import eu.fbk.soa.process.domain.DomainObject;
+
+
+@XmlAccessorType(XmlAccessType.NONE)
+public abstract class ProcessNode {
+
+	private static int nodeIndex = 0; 
+	
+	int nodeNr;
+	
+	@XmlAttribute
+	private String nodeID = "";
+	
+	public ProcessNode() {
+		this.nodeNr = nodeIndex;
+		nodeIndex++;
+	}
+	
+	public Set<DomainObject> getRelatedDomainObjects() {
+		return new HashSet<DomainObject>();
+	}
+
+	public String getNodeID() {
+		if (nodeID == "") {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.